This notebook helps us to load fMRI brain scans from Neurovault and fetch Cognitive Concepts from Cognitive Atlas 

In [2]:
!pip install nilearn

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.3/10.3 MB 87.6 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.3/3.3 MB 87.9 MB/s eta 0:00:00
  Attempting uninstall: nibabel
    Found existing installation: nibabel 3.0.2
    Uninstalling nibabel-3.0.2:
      Successfully uninstalled nibabel-3.0.2


In [19]:
from nilearn.datasets import fetch_neurovault
from urllib.request import urlopen
import json


Fetching

In [5]:
MAX_IMAGES = 5
fmri_scans = fetch_neurovault(max_images=MAX_IMAGES)

Reading local neurovault data.
Already fetched 1 image
Already fetched 2 images
Already fetched 3 images
Already fetched 4 images
Already fetched 5 images
5 images found on local disk.


In [7]:
fmri_scans

{'images': ['/root/nilearn_data/neurovault/collection_79/image_11021.nii.gz',
  '/root/nilearn_data/neurovault/collection_79/image_11020.nii.gz',
  '/root/nilearn_data/neurovault/collection_4008/image_65426.nii.gz',
  '/root/nilearn_data/neurovault/collection_1257/image_17327.nii.gz',
  '/root/nilearn_data/neurovault/collection_1257/image_17328.nii.gz'],
 'images_meta': [{'url': 'http://neurovault.org/images/11021/',
   'id': 11021,
   'file': 'http://neurovault.org/media/images/79/neurosynth_spectral_01_1.nii.gz',
   'collection': 'http://neurovault.org/collections/79/',
   'collection_id': 79,
   'file_size': 72320,
   'cognitive_paradigm_cogatlas': None,
   'cognitive_paradigm_cogatlas_id': None,
   'cognitive_contrast_cogatlas': None,
   'cognitive_contrast_cogatlas_id': None,
   'map_type': 'other',
   'analysis_level': None,
   'name': 'spectral 02',
   'description': None,
   'add_date': '2016-01-21T18:22:52.850294+01:00',
   'modify_date': '2016-01-27T22:45:17.067551+01:00',
  

Now fetching from Cognitive Atlas

In [22]:
cog_concepts = urlopen('https://www.cognitiveatlas.org/api/v-alpha/concept')
cog_concepts = json.loads(cog_concepts.read())

In [24]:
len(cog_concepts)

886

We just fetched 886 cognitive concepts. It is upto us which ones to map the neurovault images to.

In [23]:
cog_concepts

[{'id_concept_class': 'ctp_C3',
  'creation_time': 1512660626063,
  'event_stamp': '2009-06-22T19:12:54',
  'def_id_user': 'usr_4e42a3196fbcd',
  'def_event_stamp': '2011-08-10T20:41:26',
  'last_updated': 1512660626063,
  'def_id': 'def_4e42ecf5ef5a9',
  'name': 'abductive reasoning',
  'alias': '',
  'definition_text': 'The process of adopting an explanatory hypothesis; inferring the cause A as a possible explanation for the consequence B.',
  'id_user': 'usr_0000000000',
  'id': 'trm_4a3fd79d096be'},
 {'id_concept_class': 'ctp_C3',
  'creation_time': 1512660626088,
  'event_stamp': '2009-06-22T19:12:54',
  'def_id_user': 'usr_49a467bf4e0db',
  'def_event_stamp': '2012-03-28T15:36:11',
  'last_updated': 1512660626088,
  'def_id': 'def_4f732feb3ba80',
  'name': 'abstract analogy',
  'alias': '',
  'definition_text': 'high-level analogy that retains general information relevant to many specific instances',
  'id_user': 'usr_0000000000',
  'id': 'trm_4a3fd79d096e3'},
 {'id_concept_class